src/materials/EffectiveStressMaterial.C

```diff
@@ -0,0 +1,82 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "EffectiveStressMaterial.h"
+#include "EffectiveStressTools.h"
+#include "InterfaceValueTools.h"
+
+registerMooseObject("DeerApp", EffectiveStressMaterial);
+
+template <> InputParameters validParams<EffectiveStressMaterial>() {
+  InputParameters params = validParams<Material>();
+  params.addClassDescription("Compute an effective at each quadrature point");
+  params.addRequiredParam<MooseEnum>("effective_stress_type",
+                                     EffectiveStressTools::scalarOptions(),
+                                     "Type of effective stress to be computed");
+  params.addParam<std::vector<Real>>("params_vector",
+                                     "vector of effective stress parameters");
+  params.addParam<Point>("point1", Point(0, 0, 0),
+                         "Start point for axis used to calculate some "
+                         "cylindrical material tensor quantities");
+  params.addRequiredParam<MaterialPropertyName>(
+      "effective_stress_mp_name", "the name of the new  material_property");
+  params.addParam<Point>(
+      "point2", Point(0, 1, 0),
+      "End point for axis used to calculate some material tensor quantities");
+  params.addParam<Point>("direction", Point(0, 0, 1), "Direction vector");
+  params.addParam<bool>(
```

Comment:
Can you take this decision out of their hands?

Reply:
if you are taking about making the material property stateful moose should do it automatically if requested somewhere else, but apparently it doesn't work in this case.